js/apps/system/_admin/aardvark/APP/react/src/views/query/editor/BindVariablesTab.tsx

```diff
@@ -0,0 +1,141 @@
+import {
+  Box,
+  Flex,
+  Input,
+  Table,
+  TableContainer,
+  Tag,
+  Tbody,
+  Td,
+  Text,
+  Th,
+  Thead,
+  Tr
+} from "@chakra-ui/react";
+import { ValidationError } from "jsoneditor-react";
+import React from "react";
+import { ControlledJSONEditor } from "../../../components/jsonEditor/ControlledJSONEditor";
+import { JSONErrors } from "../../../components/jsonEditor/JSONErrors";
+import { useQueryContext } from "../QueryContextProvider";
+
+const detectType = (value: string) => {
+  try {
+    const parsedValue = JSON.parse(value);
+    if (parsedValue instanceof Array) {
+      return "array";
+    }
+    return typeof parsedValue;
+  } catch (ignore) {}
+};
+const parseInput = (value: string) => {
+  try {
+    return JSON.parse(value);
+  } catch (e) {
+    return value;
+  }
+};
+const DEFINED_TYPES = ["string", "number", "array", "object", "boolean"];
+export const BindVariablesTab = ({ mode }: { mode: "json" | "table" }) => {
+  const { queryBindParams, onBindParamsChange, bindVariablesJsonEditorRef } =
+    useQueryContext();
+  const [errors, setErrors] = React.useState<ValidationError[]>([]);
+  if (mode === "table") {
+    return (
+      <TableContainer>
+        <Table size="sm">
+          <Thead>
+            <Tr>
+              <Th>Key</Th>
+              <Th>Value</Th>
+            </Tr>
+          </Thead>
+          <Tbody>
+            {Object.keys(queryBindParams).map(key => {
+              return <BindVariableRow key={key} bindParamKey={key} />;
+            })}
+            {Object.keys(queryBindParams).length === 0 ? (
+              <Tr>
+                <Td colSpan={2}>No bind parameters found in query.</Td>
+              </Tr>
+            ) : null}
+          </Tbody>
+        </Table>
+      </TableContainer>
+    );
+  }
+  return (
+    <Box position="relative" height="full">
+      <ControlledJSONEditor
+        ref={bindVariablesJsonEditorRef}
+        mode="code"
+        defaultValue={queryBindParams}
+        onChange={updatedValue => {
+          !errors.length && onBindParamsChange(updatedValue || {});
+        }}
+        htmlElementProps={{
+          style: {
+            height: "calc(100% - 20px)"
+          }
+        }}
+        onValidationError={errors => {
+          setErrors(errors);
+        }}
+        /**
+         * This option will get passed on the jsonEditor.
+         * */
+        // @ts-ignore
+        mainMenuBar={false}
```

Comment:
Can we include this element at the interface so it is not TS ignored?

Reply:
not possible due to the underlying library